python/pyiceberg/cli/console.py

```diff
@@ -0,0 +1,228 @@
+        if type != "any":
+            output.exception(exc)
+
+    found_table = False
+    # We expect a namespace
+    if len(identifier_tuple) > 1:
+        try:
+            table = catalog.load_table(identifier_tuple)
+            assert table.metadata
+            output.describe_properties(table.metadata.properties)
+            found_table = True
+        except NoSuchTableError as exc:
+            if type != "any":
+                output.exception(exc)
+
+    if type == "any" and not found_namespace and not found_table:
+        output.exception(NoSuchNamespaceError(f"Could not find table/namespace with identifier {identifier}"))
+
+
+@run.command()
+@click.argument("identifier")
+@click.pass_context
+def schema(ctx, identifier: str):
+    catalog, output = _get_catalog_and_output(ctx)
+    metadata = catalog.load_table(identifier).metadata
+    output.schema(metadata.current_schema())
+
+
+@run.command()
+@click.argument("identifier")
+@click.pass_context
+def spec(ctx, identifier: str):
+    catalog, output = _get_catalog_and_output(ctx)
+    metadata = catalog.load_table(identifier).metadata
+    output.spec(metadata.current_partition_spec())
+
+
+@run.command()
+@click.argument("identifier")
+@click.pass_context
+def uuid(ctx, identifier: str):
+    catalog, output = _get_catalog_and_output(ctx)
+    metadata = catalog.load_table(identifier).metadata
+    output.uuid(metadata.table_uuid)
+
+
+@run.command()
+@click.argument("identifier")
+@click.pass_context
+def location(ctx, identifier: str):
+    catalog, output = _get_catalog_and_output(ctx)
+    metadata = catalog.load_table(identifier).metadata
+    output.text(metadata.location)
+
+
+@run.group()
+def drop(ctx):
+    pass
+
+
+@drop.command()
+@click.argument("table")
+@click.pass_context
+def table(ctx, table: str):
+    """Drop table"""
+    catalog, output = _get_catalog_and_output(ctx)
+
+    try:
+        catalog.drop_namespace(table)
+        output.text(f"Dropped table: {table}")
+    except Exception as exc:
+        output.exception(exc)
+
+
+@drop.command()
+@click.argument("namespace")
+@click.pass_context
+def namespace(ctx, namespace: str):
+    """Drop namespace"""
+    catalog, output = _get_catalog_and_output(ctx)
+
+    try:
+        catalog.drop_namespace(namespace)
+        output.text(f"Dropped namespace: {namespace}")
+    except Exception as exc:
+        output.exception(exc)
+
+
+@run.command()
+@click.argument("from_table")
+@click.argument("to_table")
+@click.pass_context
+def rename_table(ctx, from_table: str, to_table: str):
+    catalog, output = _get_catalog_and_output(ctx)
+
+    try:
+        catalog.rename_table(from_table, to_table)
+        output.text(f"Table {from_table} has been renamed to {to_table}")
+    except Exception as exc:
+        output.exception(exc)
+
+
+@run.command()
+@click.argument("from_identifier")
+@click.argument("to_identifier")
+@click.pass_context
+def table(ctx, from_identifier: str, to_identifier: str):
```

Comment:
Do we need this and `rename_table`? I'd probably go with just `rename` for the command.